lib/utils/format.ts

```diff
@@ -0,0 +1,256 @@
+
+export function format(arr: unknown[], dialect: string): string {
+  const timeZone = null;
+
+  // Make a clone of the array because format modifies the passed args
+  return SqlString.format(arr[0], arr.slice(1), timeZone, dialect);
+}
+
+export function formatNamedParameters(
+  sql: string,
+  parameters: Record<string, unknown>,
+  dialect: string,
+): string {
+  return SqlString.formatNamedParameters(sql, parameters, null, dialect);
+}
+
+export type FinderOptions<TAttributes> = {
+  attributes?: string[],
+  where?: WhereOptions<TAttributes>,
+};
+
+export type MappedFinderOptions<TAttributes> = Omit<FinderOptions<TAttributes>, 'attributes'> & {
+  // an array of attribute-column mapping, or just attributes
+  attributes?: Array<[columnName: string, attributeName: string] | string>,
+};
+
+/**
+ * Expand and normalize finder options.
+ * Mutates the "options" parameter.
+ *
+ * @param options
+ * @param Model
+ */
+export function mapFinderOptions<M extends Model, T extends FinderOptions<M['_attributes']>>(
+  options: T,
+  Model: ModelStatic<Model>,
+): MappedFinderOptions<M['_attributes']> {
+  if (Array.isArray(options.attributes)) {
+    options.attributes = Model._injectDependentVirtualAttributes(
+      options.attributes,
+    );
+
+    options.attributes = options.attributes.filter(
+      v => !Model._virtualAttributes.has(v),
+    );
+  }
+
+  mapOptionFieldNames(options, Model);
+
+  return options;
+}
+
+/**
+ * Used to map field names in attributes and where conditions.
+ *
+ * Mutates the "options" parameter.
+ *
+ * @param options
+ * @param Model
+ */
+export function mapOptionFieldNames<M extends Model>(
+  options: FinderOptions<M['_attributes']>,
+  Model: ModelStatic<Model>,
+): MappedFinderOptions<M['_attributes']> {
+  const out: MappedFinderOptions<M['_attributes']> = options;
+
+  if (Array.isArray(options.attributes)) {
+    out.attributes = options.attributes.map(attributeName => {
+      // Object lookups will force any variable to strings, we don't want that for special objects etc
+      if (typeof attributeName !== 'string') {
+        return attributeName;
+      }
+
+      // Map attributes to column names
+      const columnName: string | undefined = Model.rawAttributes[attributeName]?.field;
+      if (columnName && columnName !== attributeName) {
+        return [columnName, attributeName];
+      }
+
+      return attributeName;
+    });
+  }
+
+  if (options.where != null && isPlainObject(options.where)) {
+    out.where = mapWhereFieldNames(options.where, Model);
+  }
+
+  return out;
+}
+
+export function mapWhereFieldNames(where: Record<string | symbol, any>, Model: ModelStatic<Model>): object {
+  if (!where) {
+    return where;
+  }
+
+  const newAttributes: Record<string | symbol, any> = Object.create(null);
+  // note on 'as any[]': TypeScript < 4.4 does not support using Symbol for keys.
+  // Cast can be removed in sept. 2022 when we drop support for < 4.4
```

Comment:
Maybe we can set an expiring TODO or similar for this? I was looking at [the expiring-todo-comments rule from eslint-plugin-unicorn](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/expiring-todo-comments.md) earlier, maybe we could implement that?

Reply:
We can! Are you ok with me setting it to 'warn' instead of 'error'? I don't want unrelated PRs to start failing because the date expired

Reply:
Yes, that's a good idea